src/gs-installed-page.c

```diff
@@ -546,44 +553,74 @@ gs_installed_page_has_app (GsInstalledPage *self,
 }
 
 static void
-gs_installed_page_pending_apps_changed_cb (GsPluginLoader *plugin_loader,
-                                           GsInstalledPage *self)
+gs_installed_page_pending_apps_refined_cb (GObject *source,
+					   GAsyncResult *res,
+					   gpointer user_data)
 {
-	GsApp *app;
+	GsPluginLoader *plugin_loader = GS_PLUGIN_LOADER (source);
+	GsInstalledPage *self = GS_INSTALLED_PAGE (user_data);
 	GtkWidget *widget;
-	guint i;
-	guint cnt = 0;
-	g_autoptr(GsAppList) pending = NULL;
+	guint pending_apps_count = 0;
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GsAppList) list = gs_plugin_loader_job_process_finish (plugin_loader,
+									 res,
+									 &error);
+	if (list == NULL) {
+		if (!g_error_matches (error, GS_PLUGIN_ERROR, GS_PLUGIN_ERROR_CANCELLED))
+			g_warning ("failed to refine pending apps: %s", error->message);
+		goto out;
+	}
 
-	/* add new apps to the list */
-	pending = gs_plugin_loader_get_pending (plugin_loader);
-	for (i = 0; i < gs_app_list_length (pending); i++) {
-		app = gs_app_list_index (pending, i);
+	for (guint i = 0; i < gs_app_list_length (list); ++i) {
```

Comment:
Stylistically (and for consistency with the rest of gnome-software), I prefer `i++` (reserving `++i` for those cases where the order of operation actually matters), but I can live with it as is if you feel strongly (or has already been approved with `++i` upstream).

Reply:
I got used to the `++i` with the excuse that it's more efficient if the compiler didn't optimize it, so I like to use it. Let's hear from upstream, I have rebased + pushed the PR again upstream today.

Reply:
As I understand it, all modern compilers will optimize it.  In any case, whatever upstream likes is fine with me.  I just noticed that the style appeared to post-increment elsewhere in gnome-software, so I thought it would be nice to remain consistent, but in the end I really don't care.